### PR TITLE
improvements to test_user

### PR DESCRIPTION
**Pull request by mgnielsen1**
added improvements to user_tests to reduce code duplication & improve randomization of tests

added decorator:     @pytest.mark.parametrize('exec_number', range(Config.NUM_TEST_EXECUTIONS))
to no-input tests to set them to run multiple times (set to 3 inside config.py)

added random integer for check_not_exist and get_all tests

added FakeUserInfo to reduce duplicated code

condensed test_update_user tests into single test by using @pytest.mark.parametrize decorator & sending dictionaries as inputs

removed test_user_old.py

all tests are passing

### CODE REVIEW

File: tests/test_user.py [hunk 1]
```diff
@@ -1,93 +1,82 @@
...
         assert User.exists(user.id)
 
-    def test_user_not_exist(self, test_db):
+    @pytest.mark.parametrize('exec_number', range(Config.NUM_TEST_EXECUTIONS))
+    def test_user_not_exist(self, test_db, exec_number):
         """
         check that user_id 5000 does not exist
         """
-        id = 5000
+        id = random.randint(1, 100000)
```
Comment (wrmacdonald) [inline tests/test_user.py:59]: If id was < 10 or so, wouldn't there be a chance there'd be a user with that id? We're not deleting out all the users before checking.

**Review by wrmacdonald — APPROVED** [review]
Hey, you got parameterizing working! :D

**Comment by mgnielsen1** [comment]
Yea, good point, I’ll change the range on it so it has a relatively high
minimum value

File: tests/test_user.py (hunk 1 above)
Comment (mgnielsen1) [reply tests/test_user.py:59]: ok I just cleared the user table before the operation

**Pull request merged**